File: Scripts/film-cs/CommandBuilderContext.cs
using System.Collections.Generic;

namespace FilmCS
{
    public class CommandBuilderContext
    {
        public FilmSettings Settings { get; set; }
        public string Output { get; set; }
        public int InputIndex { get; set; } = 0;
        public string CurrentLabel { get; set; } = "[base]";
        public string CurrentAudioLabel { get; set; } = null;
        public List<string> Inputs { get; set; } = new List<string>();
        public List<string> Filters { get; set; } = new List<string>();
        public List<Track> ConnectTracks { get; set; } = new List<Track>();
        public List<Track> AudioTracks { get; set; } = new List<Track>();
        public List<Track> OverlaysTracks { get; set; } = new List<Track>();
        public Track PreTrack { get; set; } = null;
        public bool Overwrite { get; set; } = true;

        public CommandBuilderContext(FilmSettings settings = null)
        {
            Settings = settings ?? new FilmSettings();
            Output = Settings.Output;
        }
    }

    public class FilterResult
    {
        public string Filter { get; set; }
        public string Label { get; set; }
        public FilterResult(string filter, string label)
        {
            Filter = filter;
            Label = label;
        }
    }
}

File: Scripts/film/commandBuilderContext.py
from filmSettings import FilmSettings

from typing import List

class CommandBuilderContext:
    """
    Context manager for command execution.
    It provides a way to execute commands with a specific context.
    """

    def __init__(self, settings: FilmSettings = None):
        self.settings = settings if settings else FilmSettings()
        self.output = settings.output

        self.input_index = 0
        self.current_label = "[base]"
        self.current_audio_label = None
        self.inputs = []
        self.filters = []

        self.connect_tracks = []
        self.audio_tracks = []
        self.overlays_tracks = []

        self.pre_track = None
        self.overwrite = True

class FilterResult:
    def __init__(self, filter: str, label: str):
        """
        Initialize the FilterResult.

        Args:
            filter_str (str): The filter string to be used in the command.
            label (str): The label for the filter.
        """
        self.filter = filter
        self.label = label